provider/marathon_test.go

```diff
@@ -551,7 +551,7 @@ func TestMarathonTaskFilter(t *testing.T) {
 					},
 				},
 			},
-			expected:         false,
+			expected:         true,
```

Comment:
If you look at end of the test case definitions, you can see what it does: For each test case, it directly calls the filter function passing our test fixtures (i.e., the application and task structs we set up) and expects the filtering to apply (or not apply) depending on the expected outcome.